examples/hallowelt/go_app/README.md

```diff
@@ -0,0 +1,253 @@
+
+```go
+func main() {
+	ccID := os.Getenv("CC_ID")
+	logger.Infof("Environment variable CC_ID: %v", ccID)
+	// get network
+	network, err = utils.SetupNetwork("mychannel")
+
+	// Get FPC Contract
+	contract := fpc.GetContract(network, ccID)
+
+	// Invoke FPC Chaincode storeAsset
+	logger.Infof("--> Invoke FPC chaincode storeAsset: ")
+	result, err := contract.SubmitTransaction("storeAsset", "asset1", "100")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+}
+)
+```
+* To be able to query the ledger we have to call the same method changing the first parameter to `retrieveAsset` and since we are querying there is no need for the third parameter.
+```go
+// Invoke FPC Chaincode retrieveAsset
+	logger.Infof("--> Invoke FPC chaincode retrieveAsset: ")
+	result, err = contract.SubmitTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+```
+* Both previous transactions have been registered on the ledger, to be able to query without this occurring we will use the `EvaluateTransaction` go method instead.
+```go
+	// Evaluate FPC Chaincode retrieveAsset
+	logger.Infof("--> Evaluate FPC chaincode retrieveAsset: ")
+	result, err = contract.EvaluateTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Evaluate transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+```
+
+The whole text of the program would be
+```go
+package main
+
+import (
+	"os"
+
+	fpc "github.com/hyperledger-labs/fabric-private-chaincode/client_sdk/go/pkg/gateway"
+	"github.com/hyperledger-labs/fabric-private-chaincode/integration/client_sdk/go/utils"
+	"github.com/hyperledger/fabric-sdk-go/pkg/gateway"
+	"github.com/hyperledger/fabric/common/flogging"
+)
+
+var (
+	network  *gateway.Network
+	contract fpc.Contract
+	err      error
+	logger   = flogging.MustGetLogger("helloworld")
+)
+
+func main() {
+	ccID := os.Getenv("CC_ID")
+	logger.Infof("Environment variable CC_ID: %v", ccID)
+	// get network
+	network, err = utils.SetupNetwork("mychannel")
+
+	// Get FPC Contract
+	contract := fpc.GetContract(network, ccID)
+
+	// Invoke FPC Chaincode storeAsset
+	logger.Infof("--> Invoke FPC chaincode storeAsset: ")
+	result, err := contract.SubmitTransaction("storeAsset", "asset1", "100")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+	
+	// Invoke FPC Chaincode retrieveAsset
+	logger.Infof("--> Invoke FPC chaincode retrieveAsset: ")
+	result, err = contract.SubmitTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+	
```

Comment:
I would remove this, since retrieveAsset is then also called using EvaluateTransaction
```suggestion

```

Reply:
but submit is not the same as Evaluate. And also, we are just executing the same transactions that are done in the HelloWorld so that they can compare one and the other

Reply:
I know, nevertheless we demonstrate SubmitTransaction already with the `storeAsset` function. I think this is good enough - and there is no need to call `retrieveAsset` using Submit and Evaluate. (Maybe the original HelloWorld example is also not good). However, we can keep it for a moment see what Bruno things. :)